cpp/oneapi/dal/backend/primitives/reducer_dpc_test.cpp

```diff
@@ -0,0 +1,171 @@
+        6.22497295e-01, 6.98835834e-02, 3.02689288e-01, 9.20716317e-01, 5.38847056e-01,
+        1.31065752e-01, 5.20964164e-02, 3.94788090e-01, 4.44441208e-01, 8.41970072e-01,
+        7.95839642e-01, 7.06222941e-01, 3.07094727e-01, 4.71635051e-01, 6.95167909e-01,
+        4.87705811e-01, 9.80875880e-01, 8.47429993e-01, 7.99278573e-01, 8.03055359e-01,
+        5.15041949e-01, 1.54601610e-01, 1.83452684e-01, 5.44926529e-01, 3.61060027e-01,
+        5.30889049e-01, 3.94774560e-01, 6.45476808e-01, 3.20248367e-01, 7.51242537e-01,
+        6.79475419e-01, 9.77114986e-02, 8.00135435e-01, 5.13456528e-01, 5.39903025e-01,
+        2.22219806e-01, 8.61253542e-02, 8.37682505e-01, 4.00777543e-01, 9.65276184e-02,
+        4.66551200e-01, 6.56479074e-01, 8.19843719e-01, 3.71762075e-01, 9.27065163e-01,
+        6.51141093e-02, 1.75910189e-01, 8.97997910e-01, 4.84584555e-01, 7.37448745e-01,
+        5.29170464e-01, 5.03419050e-01, 5.86423732e-01, 5.22511174e-01, 2.03679645e-01,
+        6.06334781e-01, 3.35332714e-01, 4.70270025e-01, 5.77608700e-01, 3.17506150e-01,
+        4.26761534e-01, 1.21746261e-01, 1.29382337e-01, 3.39261222e-01, 5.26868973e-01,
+        9.78393336e-01, 4.32418987e-01, 9.75388516e-01, 4.45526191e-01, 7.28958199e-01,
+        2.19904203e-01, 7.12695837e-01, 7.15989708e-01, 9.71994816e-01, 4.20998253e-01,
+        5.47380502e-01, 7.61885168e-01, 5.12042467e-01, 3.71666762e-01, 1.59418935e-01,
+        8.08032749e-01, 3.83869967e-01, 7.37555502e-01, 5.57652733e-01, 4.01889435e-01,
+        7.02682404e-01, 7.85953416e-01, 8.20295938e-01, 8.36124252e-01, 3.64473260e-01,
+        7.88578965e-01, 9.31515023e-02, 4.29578336e-01, 5.94997849e-01, 5.96603448e-02,
+        3.74976922e-01, 6.58367720e-01, 2.05258010e-01, 7.71357150e-01, 6.40231937e-01,
+        1.87657221e-01, 4.52644412e-01, 8.54667886e-01, 7.49524561e-01, 3.75177343e-01,
+        8.02357007e-01, 5.85784764e-01, 3.14848323e-01, 9.07061320e-01, 7.73886588e-01,
+        6.93191912e-01, 5.14800505e-01, 8.06316758e-01, 1.15278330e-01, 7.21638091e-01,
+        9.04321900e-01, 4.16006451e-01, 4.50714398e-01, 3.04517278e-01, 2.99471721e-01,
+        7.96779651e-01, 8.08662734e-01, 4.35474855e-01, 1.72980348e-01, 1.29659259e-01,
+        9.62212414e-01, 9.36481211e-01, 3.55825253e-01, 2.38375772e-01, 5.49425873e-01,
+        7.14279925e-01, 5.11857389e-01, 5.44607827e-01, 5.10422096e-01, 6.45852780e-01,
+        1.87292744e-01, 6.22012556e-01
+    };
+    const float result[ny] = { 3.77330869, 5.29384345, 4.35266895, 4.65816082, 3.67253516,
+                               3.38860884, 4.93397974, 2.75631738, 3.77672279, 5.06383597,
+                               3.15586919, 3.26941334, 4.55140545, 4.81547106, 2.89317635,
+                               3.23149481, 4.99441006, 4.67778355, 3.75376728, 6.19627221,
+                               3.38283993, 3.80625953, 3.84272132, 4.08897591, 4.75724009,
+                               4.40515521, 5.66621902, 4.3339877,  4.47339093 };
```

Comment:
Tests should be short and understandable. Please, provide shorten data that I can look and check the test correctness by my eyes.

Reply:
Okay. Maybe automatic generation is okay?